src/shared/components/cancerSummary/SummaryBarGraph.tsx

```diff
@@ -0,0 +1,215 @@
+import * as React from "react";
+import * as _ from 'lodash';
+import {computed, observable} from "mobx";
+import { ChartTooltipItem } from '@types/chart.js';
+import Chart, {ChartLegendItem} from 'chart.js';
+import {IBarGraphData} from './CancerSummary';
+import {observer} from "mobx-react";
+import './styles.scss';
+
+interface ISummaryBarGraphProps {
+    data: IBarGraphData;
+    yAxis: 'alt-freq' | 'abs-count';
+    xAxis: 'y-axis' | 'can-types';
+    legend: boolean;
+    setPngAnchor:any;
+    setPdfAnchor:any;
+}
+
+interface IChartLegendItem extends ChartLegendItem {
+    datasetIndex: number;
+}
+
+@observer
+export default class SummaryBarGraph extends React.Component<ISummaryBarGraphProps, {}> {
+
+    private chartTarget:HTMLCanvasElement;
+
+    @observable private chart:any;
+    @observable private chartConfig:any={type:'bar'};
+
+    constructor() {
+        super();
+
+        this.loadChart = this.loadChart.bind(this);
+        this.updateChart =this.updateChart.bind(this);
+    }
+
+    private getTooltipOptions(tooltipModel: any, data:IBarGraphData, chartOptions:any) {
+        // Tooltip Element
+        let tooltipEl = document.getElementById('chartjs-tooltip');
+
+        // Create element on first render
+        if (!tooltipEl) {
+            tooltipEl = document.createElement('div');
+            tooltipEl.id = 'chartjs-tooltip';
+            tooltipEl.innerHTML = "<div></div>";
+            document.body.appendChild(tooltipEl);
+        }
+
+        // Hide if no tooltip
+        if (tooltipModel.opacity === 0) {
+            tooltipEl.style.opacity = '0';
+            // tooltipEl.style.display = 'none';
+            return;
+        }
+
+        // Set caret Position
+        tooltipEl.classList.remove('above', 'below', 'no-transform');
+        if (tooltipModel.yAlign) {
+            tooltipEl.classList.add(tooltipModel.yAlign);
+        } else {
+            tooltipEl.classList.add('no-transform');
+        }
+
+        function getBody(bodyItem:any, i:number) {
+            const {label, count, percent} = data.datasets[tooltipModel.dataPoints[i].datasetIndex];
+            return {label, count, percent};
+        }
+
+        // Set Text
+        if (tooltipModel.body) {
+
+            const bodyLines:{ label:string; count:number; percent:number }[] = tooltipModel.body.map(getBody);
+            const {totalCases, altTotalPercent} = data.datasets[tooltipModel.dataPoints[0].datasetIndex];
+
+            let innerHtml = (
```

Comment:
better to write this as JSX, here and everywhere else instead of string innerHTML.

Reply:
lets discuss how/if we should make this react.  it's possible to do.

Reply:
what do you mean?

Reply:
I understand now, didn't look carefully enough. It's fine the way it is.